app/controllers/designs_controller.rb

```diff
@@ -1,2 +1,46 @@
 class DesignsController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @designs = Design.all
+  end
+
+  def show
+    @design = Design.find(params[:id])
+  end
+
+  def new
+    @design = Design.new
+  end
+
+  def create
+    @design = Design.new(design_params)
+    if @design.save
+      redirect_to design_path(@design)
+    else
+      render :new
+    end
+  end
```

Comment:
To be able to save the design, is necessary to have the user who created it. So, it will be something like this:
`c_user = current_user`
`@design.user = c_user`